File: SEMESTER 6/OOP/Assignments/WEEK4/LocalVariableSupressor.py
class Class:
    class_variable: int = 0 # class variable

    def __init__(self, variable:int): # local variable
        self.variable = 90 # instance variable
        variable = self.variable # supressed using self keyword
        print(variable)

class LocalVariableSupressor:
    @staticmethod
    def main(*args: list[str]) -> None:
        Class(100)


if __name__ == "__main__": 
    LocalVariableSupressor.main()

        
    

File: SEMESTER 6/OOP/Assignments/WEEK4/LocalVariableSupressor.java
class Class {
    int variable = 90; // instance variable
    static int class_variable = 0; // class variable

    Class(int variable) { // local variable
        variable = this.variable; // supressed using this keyword
        System.out.println(variable);
    }
}

public class LocalVariableSupressor {
    public static void main(String[] args) {
        new Class(100);
    }
}
